src/main/scala/akka/wamp/Wamp.scala

```diff
@@ -52,7 +61,12 @@ object Wamp extends ExtensionId[WampExtension] with ExtensionIdProvider {
     * @param url is the URI to connect to (e.g. "ws://somehost.com:9999/path/to/ws")
     * @param subprotocol is the WebSocket subprotocol to negotiate (e.g. "wamp.2.msgpack" or  "wamp.2.json")
     */
-  final case class Connect(client: ActorRef, url: String = "ws://localhost:8080/ws", subprotocol: String = "wamp.2.json") extends Command
+  final case class Connect(client: ActorRef,
+                           url: String = "ws://localhost:8080/ws",
+                           subprotocol: String = "wamp.2.json",
+                           messageFlowDecider: Supervision.Decider = {
```

Comment:
Thanks for this! It looks a very good idea as I can also read it from:

http://doc.akka.io/docs/akka/2.4.9/scala/stream/stream-error.html

Unfortunately, I cannot merge your changes as they stand because the JsonSerializing code evolved to something quite different. It makes use of Scalactic disjunctions to prevent exceptions to be thrown on deserialize but rather values to be returned (in pure functional programming style)

I'll try to do my best to include a Supervision anyway